### PR TITLE
CR-1258 logging

### PR DESCRIPTION
**Pull request by akritisinhaa**
1) Logging levels have been updated to align with the RestExceptionHandler and as appropriate
2) Logging changes made which would help to create dashboards for errors/warnings in gcp

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java [hunk 1]
```diff
@@ -61,7 +61,7 @@ public CaseDTO getLatestValidNonHICaseByUPRN(final UniquePropertyReferenceNumber
           .debug("non HI latest valid case retrieved for UPRN");
       return mapperFacade.map(caseFound.get(), CaseDTO.class);
     } else {
-      log.debug("No cases returned for uprn: " + uprn);
+      log.with("uprn", uprn).warn("No cases returned for uprn");
```
Comment (andrewjohnysons) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java:64]: Not sure that these should be warnings. At best info in my opinion.
Just hasn't found any cases - woopey doo - who's going to read that?
That's just my personal opinion

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public CaseDTO modifyAddress(final AddressChangeDTO addressChanges) throws CTPEx
     Optional<CollectionCase> caseMatch = dataRepo.readCollectionCase(caseId);
 
     if (caseMatch.isEmpty()) {
-      log.with("caseId", caseId).error("Failed to retrieve Case from storage");
+      log.with("caseId", caseId).warn("Failed to retrieve Case from storage");
```
Comment (andrewjohnysons) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java:107]: again - why not just info

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java [hunk 3]
```diff
@@ -138,7 +138,7 @@ private CaseDTO createModifiedAddressCaseDetails(
     if (!caseData.getAddress().getUprn().equals(addressChanges.getAddress().getUprn())) {
       log.with("caseId", caseId)
           .with("uprn", addressChanges.getAddress().getUprn().toString())
-          .error("The UPRN of the referenced Case and the provided Address UPRN must be matching");
+          .warn("The UPRN of the referenced Case and the provided Address UPRN must be matching");
```
Comment (andrewjohnysons) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java:141]: Again - here -

**Review by andrewjohnysons — CHANGES_REQUESTED** [review]
CaseServiceImpl - line 108 I'd add the caseID to th exception to identify it, 
 add something to that CTPException on line 108 which is why I am requesting changes :-)

**Comment by akritisinhaa** [comment]
@andrewjohnysons made changes to line 108 as suggested

**Review by andrewjohnysons — APPROVED** [review]

**Review by devenrawat — APPROVED** [review]

**Review by rpcatling — CHANGES_REQUESTED** [review]
Just a couple of small additions. There are two calls to the `error` logging method (in the case and UAC receiver code). Please can we add the exception parameter for a stack trace.

**Comment by akritisinhaa** [comment]
@rpcatling added the stack trace and also updated the pom to point to latest framework version

**Review by rpcatling — APPROVED** [review]

**Pull request merged**